avocado/core/runner.py

```diff
@@ -513,12 +513,12 @@ def _template_to_factory(template, variant):
                          followed by parameters to the class
         :type template: tuple
         :param variant: variant to be applied, usually containing
-                        the keys: mux_path, variant and variant_id
+                        the keys: paths, variant and variant_id
         :type variant: dict
         :return: tuple(new_test_factory, applied_variant)
         """
         var = variant.get("variant")
-        mux_path = variant.get("mux_path")
+        paths = variant.get("paths")
```

Comment:
This breaks the backward compatibility. Please check for `mux_path` presence at least until 52LTS is deprecated.

Reply:
You're missing the point regarding LTS.  This branch (master) does *not* have to keep compatibility with the LTS branch (52lts), not even with itself in core modules (as in previous versions).

What we won't do is break compatibility *within* the LTS branch (52lts) and with the test APIs in the master branch (so user's tests won't break on newer versions 57.0, 58.0, etc).

Reply:
The way I understand LTS is that users should be using this and also be able to experiment with master as that is to become next LTS. Then when new LTS is released they should seamlessly migrate there. So master does not need to read 32lts results, but it should be able to read 52lts results as the next LTS should be able to view previous LTS versions and that is how 52lts worked (it's able to read 32 results).

Reply:
@clebergnu I see @ldoktor point here, it's related to the jobdata backward compatibility. We are storing the serialized variants and changing the names will avoid a further version of replaying the job created by a previous version. We have to agree on an approach for that, because last LTS was very painful to make compatible with the LTS before that. We should either have some migration tool, tracking and taking care of the compatibility breakages, or keep the compatibility code from one LTS all the way to the other. I'd prefer the former (and I just saw the card requesting that... +1 for it) and that why I accepted this code breaking the compatibility.

Reply:
We did go through extra steps to ensure that LTS data was readable by LTS+1.  But, AFAICT, it's not something we promised upfront in the definition of what LTS is.  I can go through the RFC again, but I'm pretty sure the extra 6 months of overlapping maintenance is exactly for that reason: users can evaluate the incompatibilities, work around them and then safely migrate to the new version.

What I don't want here, is to hold back further progress for the sake of "almost eternal" compatibility worries.